File: 412.fizz-buzz.java
/*
 * @lc app=leetcode id=412 lang=java
 *
 * [412] Fizz Buzz
 */

// @lc code=start

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> ret = new ArrayList<>();
        for (int ix = 1; ix != n + 1; ++ix) {
            if (ix % 15 == 0) ret.add("FizzBuzz");
            else if (ix % 3 == 0) ret.add("Fizz");
            else if (ix % 5 == 0) ret.add("Buzz");
            else ret.add(String.valueOf(ix));
        }
        return ret;
    }
}
// @lc code=end

File: 412.fizz-buzz.cpp
/*
 * @lc app=leetcode id=412 lang=cpp
 *
 * [412] Fizz Buzz
 */

// @lc code=start
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ret(n);
        for (unsigned ix = 1ull; ix <= n; ++ix) {
            if (ix % 3 == 0 && ix % 5 == 0) {
                ret[ix - 1] = "FizzBuzz"; continue;
            }
            if (ix % 3 == 0) {
                ret[ix - 1] = "Fizz"; continue;
            }
            if (ix % 5 == 0) {
                ret[ix - 1] = "Buzz"; continue;
            }
            ret[ix - 1] = to_string(ix);
        }
        return ret;
    }
};
// @lc code=end
